ports/stm32/machine_i2c.c

```diff
@@ -105,18 +112,20 @@ STATIC void machine_hard_i2c_print(const mp_print_t *print, mp_obj_t self_in, mp
 }
 
 void machine_hard_i2c_init(machine_hard_i2c_obj_t *self, uint32_t freq, uint32_t timeout) {
-    (void)timeout;
     i2c_init(self->i2c, self->scl, self->sda, freq);
+    if (self->us_timeout != NULL) {
```

Comment:
I don't think it's necessary to test for NULL, if the I2C is created it should exist and have a valid pointer (and there's no corresponding test in the read from this value below, and it's not needed there).